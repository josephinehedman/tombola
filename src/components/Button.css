.btn-get-activity {
    background-color:  #8e8cc2;
    padding: 0.5rem 2rem;
    border-radius: 0.5rem;
    border: none;
    color: white;
    width: 50vw;
    max-width: 220px;
    font-size: 1.1rem;
    font-family: 'Work Sans', sans-serif;
}

@media all and (min-width:840px) {

    .btn-get-activity {
        font-size: 2rem;
        padding: 1rem;
        max-width: 520px;
     }
  }