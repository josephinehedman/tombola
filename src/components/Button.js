import { useHistory } from 'react-router-dom';
import './Button.css';

const Button = ({ getData, participants }) => {
  const history = useHistory();

  return (
    <button className="btn btn-get-activity"onClick={() => {
      history.push('/activity');
      getData(participants || 'random');
    }}>
      Get Activity
    </button>
  );
};

export default Button;
