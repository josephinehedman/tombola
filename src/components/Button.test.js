import renderer from 'react-test-renderer';
import Button from './Button'

describe('Button component', () => {
    test('should render the button component and match snapshot', () => {
        const participants = 2;
        const getData = jest.fn();
        const button = renderer.create(<Button />).toJSON();
        expect(button).toMatchSnapshot();
    });
});