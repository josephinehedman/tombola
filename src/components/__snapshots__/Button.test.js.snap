// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render the button component and match snapshot 1`] = `
<button
  className="btn btn-get-activity"
  onClick={[Function]}
>
  Get Activity
</button>
`;
